samza-core/src/main/java/org/apache/samza/coordinator/ClusterMembership.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator;
+
+/**
+ * Coordination Primitive to maintain the list of processors in the quorum
```

Comment:
Can you add more docs to this interface? I'd like to specifically see 
* Expectation around the register and unregister methods. 
* Read after write guarantee
* Since there isn't any listeners what is required from the implementors in case of disconnects or loss of membership.
* Thread safety and concurrent access